starlite/utils/compat.py

```diff
@@ -43,8 +43,8 @@ def py_39_safe_annotations(annotated: Any) -> Generator[Any, None, None]:
     Yields:
```

Comment:
Not really related to this PR, but I recently ran into issue with this function because it only handled `Union`, and not the built-in subscriptable types. This causes route handlers defined with them to error out when used in a lower python version and with `__future__.annotations`. What do you think about handling those cases as well?

If not, could you elaborate why we only handle this specific case? (I can't recall anymore what the exact reason was you added this, sorry :grimacing:)

Reply:
I regret having implemented it and want to remove it - its trying to do too much.

Reply:
Maybe we can remove it then? Or narrow its scope somewhat?

Reply:
PR up to remove it